crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -1,52 +1,133 @@
-use common_enums::enums;
-use common_utils::types::StringMinorUnit;
+use common_enums::enums::CaptureMethod;
+use common_utils::types::MinorUnit;
 use hyperswitch_domain_models::{
     payment_method_data::PaymentMethodData,
-    router_data::{ConnectorAuthType, RouterData},
+    router_data::{AccessToken, ConnectorAuthType, RouterData},
     router_flow_types::refunds::{Execute, RSync},
-    router_request_types::ResponseId,
+    router_request_types::{PaymentsCancelData, ResponseId},
     router_response_types::{PaymentsResponseData, RefundsResponseData},
-    types::{PaymentsAuthorizeRouterData, RefundsRouterData},
+    types::{
+        PaymentsAuthorizeRouterData, PaymentsCancelRouterData, PaymentsCaptureRouterData,
+        RefreshTokenRouterData, RefundsRouterData,
+    },
 };
 use hyperswitch_interfaces::errors;
 use masking::Secret;
 use serde::{Deserialize, Serialize};
+use strum::Display;
 
 use crate::{
     types::{RefundsResponseRouterData, ResponseRouterData},
-    utils::PaymentsAuthorizeRequestData,
+    utils::{
+        get_unimplemented_payment_method_error_message, PaymentsAuthorizeRequestData,
+        RouterData as OtherRouterData,
+    },
 };
-
-//TODO: Fill the struct with respective fields
 pub struct JpmorganRouterData<T> {
-    pub amount: StringMinorUnit, // The type of amount that a connector accepts, for example, String, i64, f64, etc.
+    pub amount: MinorUnit,
     pub router_data: T,
 }
 
-impl<T> From<(StringMinorUnit, T)> for JpmorganRouterData<T> {
-    fn from((amount, item): (StringMinorUnit, T)) -> Self {
-        //Todo :  use utils to convert the amount to the type of amount that a connector accepts
+impl<T> From<(MinorUnit, T)> for JpmorganRouterData<T> {
+    fn from((amount, item): (MinorUnit, T)) -> Self {
         Self {
             amount,
             router_data: item,
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, PartialEq)]
+#[derive(Debug, Clone, Serialize)]
+pub struct JpmorganAuthUpdateRequest {
+    pub grant_type: String,
+    pub scope: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JpmorganAuthUpdateResponse {
+    pub access_token: Secret<String>,
+    pub scope: String,
+    pub token_type: String,
+    pub expires_in: i64,
+}
+
+impl TryFrom<&RefreshTokenRouterData> for JpmorganAuthUpdateRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &RefreshTokenRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            grant_type: String::from("client_credentials"),
+            scope: String::from("jpm:payments:sandbox"),
```

Comment:
Lets not hardcode this

Reply:
we don't have the prod url for access token, so there's no other option but to hardcode this rn